### ISSUE TITLE
UITableView's scrollview loses userInteraction for brief periods while using sd_imageTransition

### ISSUE THREAD

**Issue by annjawn** [rs/SDWebImage#2310]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios (Swift 4.2)
 Platform Version        | 11.0 / 10.3
 SDWebImage Version      | 4.3.3
 Integration Method      | cocoapods
 Xcode Version           | Xcode 9
 Repro rate              | all the time (100%)

### Issue Description and Steps

I have a `UITableView` with custom `UITableViewCells` which contains `UIImageView` and I am using `imageView.image1.sd_setImage(with: URL(string: myURL, placeholderImage: UIImage(named: "nophoto"), completed: nil)`. I am also using `sd_imageTransition` like so - `imageView.sd_imageTransition = .fade`.

So the code looks like this-

```
imageView.sd_imageTransition = .fade
imageView.image1.sd_setImage(with: URL(string: myURL, placeholderImage: UIImage(named: "nophoto"), completed: nil)
```

It all works great, however I am facing a weird issue where it seems like the `UITableView`'s scroll view is getting its `userInteraction` disabled for a brief periods (for example, it wouldn't respond to touch and simply keep scrolling past, especially if you begin dragging on the UIImageView) when I am fast scrolling and the image(s) transition is happening. I have tested this in real devices and it is consistently happening. if I remove the `imageView.sd_imageTransition = .fade` bit of code, it works all good. But with the transition set, this intermittent disabling of `userInteraction` is a confusing behavior for the user. Documentation doesn't seem to talk much about `sd_imageTransition`.. Am I missing something here?

**Comment by dreampiggy** [comment]
@ann
For iOS, this view transition just use the correspond `UIViewAnimationOptions`. By default, there are not set with `UIViewAnimationOptionAllowUserInteraction`, so the user interaction will be disabled during the transition.

You can simply add it by append `UIViewAnimationOptionAllowUserInteraction` to `transition.animationOptions`. For Swift user maybe using `insert(.allowUserInteraction)` is OK.

Maybe we can discuss to add this options as well ? Should we by default to allow user interaction ?

**Comment by annjawn** [comment]
By the way I am going to give it a try setting the `sd_imageTransition?.animationOptions` with `.allowUserInteraction` right now.

**Comment by annjawn** [comment]
Yup, it has worked perfectly. But I had to do `animationOptions.insert(.allowUserInteraction)` at too many places 😊

**Comment by dreampiggy** [comment]
@annjawn See #2315

**Comment by bpoplauschi** [comment]
Closed per #2315
